homeassistant/components/home_connect/__init__.py

```diff
@@ -138,11 +138,18 @@ async def _async_service_program(call, method):
         """Execute calls to services taking a program."""
         program = call.data[ATTR_PROGRAM]
         device_id = call.data[ATTR_DEVICE_ID]
-        options = {
-            ATTR_KEY: call.data.get(ATTR_KEY),
-            ATTR_VALUE: call.data.get(ATTR_VALUE),
-            ATTR_UNIT: call.data.get(ATTR_UNIT),
-        }
+
+        options = []
+
+        option_key = call.data.get(ATTR_KEY)
+        if option_key is not None:
+            option = {ATTR_KEY: option_key, ATTR_VALUE: call.data.get(ATTR_VALUE)}
+
+            option_unit = call.data.get(ATTR_UNIT)
+            if option_unit is not None:
+                setattr(option, ATTR_UNIT, option_unit)
+
+            options.append(option)
```

Comment:
We only ever append a single item to this list. We could just create the list with this item directly. We don't need to append the item.

Reply:
Is there a big difference between appending and overwriting? As the empty list of options should still be the fallback if `option_key` is not set. I thought reusing the existing list would be more efficient.

Reply:
This code makes sense now that I see that option_key is optional.